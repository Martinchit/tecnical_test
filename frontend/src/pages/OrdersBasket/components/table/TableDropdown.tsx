import React from 'react';
import { Dropdown } from 'react-bootstrap';
import { TableDropdownButton } from './TableDropdownButton';
import { OrderExecutionMode } from '../../../../types/types';
import { ExecutionMode } from '../../../../core/lib/utils/executionMode';

interface TableDropdownProps {
  title: OrderExecutionMode | undefined;
  dropdownID: string;
  onSelect: (id: number) => void;
  disabled: boolean;
}

export const TableDropdown: React.FC<TableDropdownProps> = ({
  title,
  dropdownID,
  onSelect,
  disabled,
}) => (
  <Dropdown onSelect={(id: string) => onSelect(Number(id))}>
    <TableDropdownButton
      className="pull-right"
      disabled={disabled}
      variant="light"
      id={dropdownID}
    >
      {title}
    </TableDropdownButton>
    <Dropdown.Menu>
      {ExecutionMode.map((option, i) => (
        <Dropdown.Item eventKey={`${i}`} key={option}>
          {option}
        </Dropdown.Item>
      ))}
    </Dropdown.Menu>
  </Dropdown>
);
