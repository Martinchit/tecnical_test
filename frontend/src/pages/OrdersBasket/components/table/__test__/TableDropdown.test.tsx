import React from 'react';
import { shallow } from 'enzyme';

import { TableDropdown } from '../TableDropdown';

describe('TableDropdown', () => {
  const props = {
    title: undefined,
    dropdownID: 'testDropdown',
    onSelect: jest.fn(),
    disabled: false,
  };

  const restoreMock = () => {
    props.onSelect.mockRestore();
  };

  it('renders correctly', () => {
    const wrapper = shallow(<TableDropdown {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers onSelect when a dropdown option is selected', () => {
    const optionID = 1;
    const wrapper = shallow(<TableDropdown {...props} />);
    wrapper.simulate('select', optionID);
    expect(props.onSelect).toHaveBeenCalled();
    expect(props.onSelect).toHaveBeenCalledWith(optionID);
  });
});
