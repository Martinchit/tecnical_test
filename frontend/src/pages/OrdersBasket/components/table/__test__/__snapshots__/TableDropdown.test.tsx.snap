// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableDropdown renders correctly 1`] = `
<Dropdown
  navbar={false}
  onSelect={[Function]}
>
  <Styled(DropdownToggle)
    className="pull-right"
    disabled={false}
    id="testDropdown"
    variant="light"
  />
  <DropdownMenu
    alignRight={false}
    flip={true}
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      disabled={false}
      eventKey="0"
      key="Market"
    >
      Market
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      disabled={false}
      eventKey="1"
      key="Limit"
    >
      Limit
    </DropdownItem>
  </DropdownMenu>
</Dropdown>
`;
